### ISSUE TITLE
Webpack problem

### ISSUE THREAD

**Issue by HJoost** [telerik/kendo-angular2#16]
Hi,

after long time waiting for the Angular2 Kendo UI Beta, we just try to add it to our project. But it seems that there is a problem, while building with webpack. The console of the browser shows the following error:

Unexpected value 'ButtonsModule' imported by the module 'AppModule' (compiler.umd.js:14126)

We also tried the quickstart from this link:

https://github.com/telerik/kendo-angular2-quickstart

As soon as we converted the sample project to build by webpack, the same error occurs. Does anybody have an idea what to do?

Thanks for your replies!

**Comment by HansJakob** [comment]
I have the exact same problem.

I also use webpack, and I have made my project from Angular 2's tutorial on the page:
https://angular.io/docs/ts/latest/guide/webpack.html

I also get the error:
"Uncaught Error: Unexpected value 'ButtonsModule' imported by the module 'AppModule'"

I really hope there is someone who can help.

**Comment by RobRaesemann** [comment]
I'm getting the same error using the latest agular-cli@webpack...

> Uncaught Error: Unexpected value 'GridModule' imported by the module 'AppModule'

Here is my app.module.ts...

```
import { BrowserModule } from '@angular/platform-browser';
import { NgModule } from '@angular/core';
import { FormsModule } from '@angular/forms';
import { HttpModule } from '@angular/http';
import { GridModule } from '@progress/kendo-angular-grid';

import { AppComponent } from './app.component';

@NgModule({
  declarations: [
    AppComponent
  ],
  imports: [
    BrowserModule,
    FormsModule,
    HttpModule,
    GridModule
  ],
  providers: [],
  bootstrap: [AppComponent]
})
export class AppModule { }
```

I did a little research, and there seems to be some controversy over the fact that the webpack configuration is embedded in the CLI and is not accessible or customizable by the user, so you are stuck with the way that they decide to do things. Given the fact that angular-cli is "the" CLI for 
Angular 2, and they are moving to webpack, it would seem that getting this working is going to be pretty important. 

I want to pass on some kudos and encouragement to the Kendo team for getting this beta out to us. Of course, we all want/need this stuff working yesterday. Kendo is my go to toolkit, and I'm itching to get as much experience with Angular 2 as I can. The little bit of research I did into this issue gives me an appreciation for what you are up against. This stuff is all evolving very quickly and I can only imagine how difficult some of the design decisions that you are trying to make have to be. Keep up the good work. We are all looking forward to the results.

**Comment by dimodi** [comment]
Hello,

To report issues caused by Kendo UI for Angular 2, please use the following process:

http://www.telerik.com/kendo-angular-ui/support/#bug-reporting

If a Plunkеr example is not the best or appropriate option (such as in this case), you can send us a link to a GitHub repository, or to a ZIP of your project. In all cases, we need a runnable isolated example that reproduces the discussed issue.

Based on the provided information, I can assume that the problem may be caused by:

- using ahead-of-time compilation (this is currently not supported)
- using an older Angular2 version
- different packages depend on different Angular2 versions

If using Angular2 final, check the Grid's package.json and verify its version is 0.3.1.

To provide additional comments on this case, we will need additional information, as requested above.

**Comment by HansJakob** [comment]
I've added a zip package with an isolated example of the problem.

Just run "npm i" and then "npm start".

Hope this is enough to identify the problem. :-)

https://1drv.ms/u/s!AmlyWqBmdQP9gp0HZoZvZdpFjnz13A

**Comment by dimodi** [comment]
@HansJakob 

The package.json file depends on Angular 2.0.0-rc.7, while the Kendo UI Button 0.10.0 requires 2.0.0 (final).

**Comment by HansJakob** [comment]
Sorry... I feel so stupid.. :-)

This worked.

**Comment by SilvatoX** [comment]
@dimodi 
I updated to Angular 2.0.0 (final) but the error still occurs...

**Comment by tsvetomir** [comment]
Can you paste the output from `npm ls @angular/core`?

**Comment by SilvatoX** [comment]
@tsvetomir  
It's working now. I just reinstalled my node modules. Thanks

angular2-webpack@1.0.0 C:\Angular2
`-- @angular/core@2.0.0

**Comment by Tenmak** [comment]
Hi, I'm having a similar problem most probably because of TypeScript or Angular2...

I was on Angular RC5 and decided to upgrade my whole package to the final version after seeing the previous posts. TypeScript is yelling  about the **combobox.component.d.ts** file :
I get a lot of TS2300 Duplicate Identifier 'readonly'.

I'm using TypeScript 1.8.10, should I upgrade to TypeScript 2.0.0 ? Or is it related to something else ?
Also, I'm using VS2015 RC3. I didn't have these errors when I was using the 0.9.0 packages of kendo-angular-dropdowns

**Comment by tsvetomir** [comment]
The first published release was built using TypeScript 2.0 (see #14). We published patch versions as soon as we realized TypeScript requires only 1.8 to run.

We recommend using a relaxed reference such as `"@progress/kendo-angular-dropdowns": "0.x"` in your project.json file so `npm update` can pick up the latest version at any time.

I'll close this issue, but feel free to open a new one should you encounter any problems.
